src/Microsoft.AspNet.Hosting.Server.Abstractions/IHttpApplication.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http.Features;
+
+namespace Microsoft.AspNet.Hosting.Server
+{
+    /// <summary>
+    /// Represents an HttpApplication.
+    /// </summary>
+    public interface IHttpApplication<THttpContext>
+    {
+        /// <summary>
+        /// Create an HttpContext given a collection of HTTP features.
```

Comment:
The reason I left it as that was because `THttpContext` sounds highly correlated to `HttpContext` and `IHttpApplication`. I imagine that it will always stay as a wrapper around `HttpContext` anyway so I didn't update the comments. I could change it to say `THttpContext` instead but I think `TContext` would be more appropriate if we also change the generic's name.